src/coverlet.core/Instrumentation/Instrumenter.cs

```diff
@@ -99,32 +100,59 @@ private void InstrumentMethod(MethodDefinition method)
 
         private void InstrumentIL(MethodDefinition method)
         {
+            method.Body.SimplifyMacros();
             ILProcessor processor = method.Body.GetILProcessor();
 
             var index = 0;
             var count = processor.Body.Instructions.Count;
 
+            var branchPoints = CecilSymbolHelper.GetBranchPoints(method);
+
             for (int n = 0; n < count; n++)
             {
                 var instruction = processor.Body.Instructions[index];
                 var sequencePoint = method.DebugInformation.GetSequencePoint(instruction);
-                if (sequencePoint == null || sequencePoint.IsHidden)
+                var targetedBranchPoints = branchPoints.Where(p => p.EndOffset == instruction.Offset);
+                
+                if (sequencePoint != null && !sequencePoint.IsHidden)
                 {
-                    index++;
-                    continue;
-                }
+                    var target = AddInstrumentationCode(method, processor, instruction, sequencePoint);
+                    foreach (var _instruction in processor.Body.Instructions)
+                        ReplaceInstructionTarget(_instruction, instruction, target);
 
-                var target = AddInstrumentationCode(method, processor, instruction, sequencePoint);
-                foreach (var _instruction in processor.Body.Instructions)
-                    ReplaceInstructionTarget(_instruction, instruction, target);
+                    foreach (ExceptionHandler handler in processor.Body.ExceptionHandlers)
+                        ReplaceExceptionHandlerBoundary(handler, instruction, target);
 
-                foreach (ExceptionHandler handler in processor.Body.ExceptionHandlers)
-                    ReplaceExceptionHandlerBoundary(handler, instruction, target);
+                    index += 3;
+                }
+
+                if (targetedBranchPoints.Count() > 0)
```

Comment:
There doesn't seem to be a reason to check the count here. If it's empty the next loop simply won't run